src/main/java/com/minecolonies/core/entity/pathfinding/navigation/MinecoloniesAdvancedPathNavigate.java

```diff
@@ -1061,16 +1061,19 @@ public void stop()
 
     @Nullable
     @Override
-    public WaterPathResult moveToWater(final int range, final double speed, final List<Tuple<BlockPos, BlockPos>> ponds)
+    public WaterPathResult searchWater(final int range, final double speed, final List<Tuple<BlockPos, BlockPos>> ponds)
```

Comment:
I think we shouldnt have this in the navigator, the navigators job is moving the entity/following the path. If you want a result without that just do it in the AI directly